tiny_dnn/activations/elu_layer.h

```diff
@@ -0,0 +1,44 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class elu_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "elu-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) override {
+    float_t xj;
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      xj   = x[j];
+      y[j] = xj < float_t(0) ? (std::exp(xj) - float_t(1)) : xj;
+    }
+  }
+
+  void backward_activation(const vec_t &x,
+                           const vec_t &y,
+                           vec_t &dx,
+                           const vec_t &dy) override {
+    float_t yj;
+    for (serial_size_t j = 0; j < x.size(); j++) {
+      yj = y[j];
+      // dx = dy * (gradient of elu)
+      dx[j] = dy[j] * (yj > float_t(0) ? float_t(1) : (float_t(1) + yj));
```

Comment:
We can use a general rule to prefer readbility vs microptimization if it is not critical after we get timing with optimized compiler flags.